src/sshfs_mount/sshfs_mount_handler.cpp

```diff
@@ -223,23 +218,68 @@ void SSHFSMountHandler::activate_impl(ServerVariant server, std::chrono::millise
 
 void SSHFSMountHandler::deactivate_impl(bool force)
 {
-    mpl::log(mpl::Level::info, category, fmt::format("Stopping mount \"{}\" in instance '{}'", target, vm->vm_name));
+    mpl::info(category, fmt::format("Stopping mount \"{}\" in instance '{}'", target, vm->vm_name));
     QObject::disconnect(process.get(), &Process::error_occurred, nullptr, nullptr);
-    if (process->terminate(); !process->wait_for_finished(5000))
+
+    try
     {
-        auto err = fmt::format("Failed to terminate SSHFS mount process: {}", process->read_all_standard_error());
-        if (force)
-            mpl::log(
-                mpl::Level::warning, category,
-                fmt::format("Failed to gracefully stop mount \"{}\" in instance '{}': {}", target, vm->vm_name, err));
-        else
-            throw std::runtime_error{err};
+        if (process->terminate(); !process->wait_for_finished(5000))
+        {
+            auto err = fmt::format("Failed to terminate SSHFS mount process: {}", process->read_all_standard_error());
+            if (force)
+                mpl::warn(category,
+                          "Failed to gracefully stop mount \"{}\" in instance '{}': {}",
+                          target,
+                          vm->vm_name,
+                          err);
+            else
+                throw std::runtime_error{err};
+        }
+    }
+    catch (...)
+    {
+        // The finished signal should've triggered by now. We can disconnect it.
+        // Doing it this way to ensure that it gets disconnected even when the body
+        // throws.
+        QObject::disconnect(process.get(), &Process::finished, nullptr, nullptr);
+        throw;
     }
     process.reset();
 }
 
 SSHFSMountHandler::~SSHFSMountHandler()
 {
     deactivate(/*force=*/true);
+
+    /**
```

Comment:
Did you really mean for this to be a documentation-style comment?
```suggestion
    /*
```

Reply:
It's a remnant from the documentation plugin I use (doxdocgen). I'll remove it.

Reply:
One thing I noticed is my editor stops highlighting certain keywords (i.e. FIXME) when it's not `/**`. Nevertheless, nothing major. I can live with that.

Reply:
Done

Reply:
Oh, that's annoying :confused: is there a setting for that perhaps? For example, in CLion I have settings for custom TODO/FIXME highlights to appear in different ways. Or perhaps a plugin in VSCode?

Reply:
There probably is. I recall using a color-coded keyword highlighter in the past, I'll look into it :)